digest/src/dyn_digest.rs

```diff
@@ -36,6 +49,8 @@ pub trait DynDigest {
     fn output_size(&self) -> usize;
 
     /// Clone hasher state into a boxed trait object
+    #[cfg(feature = "alloc")]
+    #[cfg_attr(docsrs, doc(cfg(feature = "alloc")))]
     fn box_clone(&self) -> Box<dyn DynDigest>;
```

Comment:
While I was able to make the two other alloc-gated methods provided, this one is notably required, and somewhat annoyingly means that downstream impls of `DynDigest` would also need to `alloc`-gate this as well.

Given that `DynDigest` is largely intended as a wrapper for `Digest` anyway, I'm not sure it's a problem.